festim/species.py

```diff
@@ -13,6 +13,8 @@ class Species:
         solution (dolfinx.fem.Function or ...): the solution for the current timestep
```

Comment:
I think we can now complete the type here, it's something like `Indexed` 
Same for line 14